### ISSUE TITLE
[🐞]: Cage can be used to exploit goblin traders

### ISSUE THREAD

**Issue by Theluckgod76** [MehVahdJukaar/Supplementaries#900]
### Before Continuing:

- [X] Always check you are using the latest version of the mods and its dependencies
- [X] Remove known problematic rendering mods such as Optifine, Sodium and others. Do NOT report issues related to said mods
- [X] If you are unsure which mod is the culprit.
Disable half your mods at time until you isolate the culprit
- [X] If you have a crash ALWAYS post LATEST.LOG below
- [X] If the issue is about behaviors be sure to consult the WIKI first

### Version-Loader

1.19.2-forge

### Supplementaries Version

supplementaries-1.19.2-2.4.11

### Moonlight Lib Version

moonlight-1.19.2-2.3.5

### Issue Detail

Not a bug, but I discovered an exploit that's made possible when you have a cage from Supplementaries, a goblin/fiery goblin from Goblin Traders, and a friend to do this with. This exploit allows you to bypass the max trade lock when trading with a goblin, resulting in a pretty op way of getting emeralds or good loot. 

### To Produce

(For this example, I will be using fiery goblins, and players A and B. Fiery goblins can trade you one emerald per carrot given with a max of around 24 I think.)
Player A has to right click on the fiery goblin to begin trading with it, and player B has to pick up the fiery goblin with their cage. Once the fiery goblin has been picked up, player A can trade the max amount of carrots and the trade will be locked off. Once player B places the fiery goblin on the ground again, the trade will not be locked off and they can redo the trades over and over. The same thing can be done with other trades.

**Comment by Xelbayria** [comment]
Interesting, that's a loophole (IIRC is the term) which considered to be a bug. that's rare.

**Comment by MehVahdJukaar** [comment]
so does this mean GUI stays open even after the goblin entiti is disposed of (meaning cage item absorbs it?)
if so kind of seems like an issue of globlin traders mod. regardless trader can be blacklisted

**Comment by Theluckgod76** [comment]
Yes the trading GUI stays open. I wasn't sure if I should mention this to Supplementaries or Goblin Traders but it was a pretty brutal exploit so I figured I should say something.

**Comment by meeprdib** [comment]
Mod versions on 1.19.2 and below are not supported
